libs/Microsoft.MixedReality.WebRTC.Native/src/api.cpp

```diff
@@ -1075,6 +1075,34 @@ void MRS_CALL mrsPeerConnectionRemoveLocalAudioTrack(
   }
 }
 
+
+
+void MRS_CALL mrsAudioReadStreamCreate(PeerConnectionHandle peerHandle,
+                                       int bufferMs,
+                                       AudioReadStreamHandle* audioBufferOut) {
+  *audioBufferOut = nullptr;
+  if (auto peer = static_cast<PeerConnection*>(peerHandle)) {
+    *audioBufferOut = new AudioReadStream(peer, bufferMs);
```

Comment:
So far in the API all objects are referenced counted and held alive by C++. The alternative here is to not reference count and use raw pointers. Both rely on the wrapper layer (C#) calling `mrsAudioReadStreamDestroy()` but the latter makes sharing and debugging much easier. Do we think that it's not worth the complication here because the audio read stream is uniquely owned by a single C# object?

Reply:
Yes, given that reads are inherently destructive, I don't expect these objects to be shared. I can ensure the handle is nulled in the C# object also to avoid double delete.